sinker/sinker.go

```diff
@@ -118,11 +118,12 @@ func (s *SQLSinker) HandleBlockScopedData(ctx context.Context, data *pbsubstream
 		}
 	}
 
-	if (s.batchBlockModulo(isLive) > 0 && data.Clock.Number%s.batchBlockModulo(isLive) == 0) || s.loader.FlushNeeded() {
+	blockFlushNeeded := s.batchBlockModulo(isLive) > 0 && data.Clock.Number-s.stats.GetLastBlockNum() >= s.batchBlockModulo(isLive)
```

Comment:
The `stats` was never meant to become "critical" for functionality which is the case now, is there any other direct data that we could use?

What also if the stats aren't yet populated, is that possible, like on resume maybe?

Reply:
Makes sense. Removed reliance on stats.